components/motor/dimensionengineering/sabertooth.go

```diff
@@ -0,0 +1,556 @@
+// Package dimensionengineering contains implementations of the dimensionengineering motor controls
+package dimensionengineering
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"math"
+	"sync"
+	"time"
+
+	"github.com/edaniels/golog"
+	"github.com/jacobsa/go-serial/serial"
+	"github.com/mitchellh/mapstructure"
+	utils "go.viam.com/utils"
+	"go.viam.com/utils/usb"
+
+	"go.viam.com/rdk/components/motor"
+	"go.viam.com/rdk/config"
+	"go.viam.com/rdk/operation"
+	"go.viam.com/rdk/registry"
+	rdkutils "go.viam.com/rdk/utils"
+)
+
+const (
+	// https://www.dimensionengineering.com/datasheets/Sabertooth2x60.pdf
+	modelName = "de-sabertooth"
+)
+
+// controllers is global to all instances, mapped by serial device.
+var (
+	globalMu    sync.Mutex
+	controllers map[string]*controller
+	usbFilter   = usb.SearchFilter{}
+)
+
+// controller is common across all Sabertooth motor instances sharing a controller.
+type controller struct {
+	mu           sync.Mutex
+	port         io.ReadWriteCloser
+	serialDevice string
+	logger       golog.Logger
+	activeAxes   map[int]bool
+	testChan     chan []byte
+	address      int // 128-135
+}
+
+// Motor is a single axis/motor/component instance.
+type Motor struct {
+	// A reference to the actual controller that needs to be commanded for the motor to run
+	c *controller
+	// which channel the motor is connected to on the controller
+	Channel int
+	// Simply indicates if the RDK _thinks_ the motor is moving, because this controller has no feedback, this may not reflect reality
+	isOn bool
+	// dirFlip means that the motor is wired "backwards" from what we expect forward/backward to mean,
+	// so we need to "flip" the direction sent by control
+	dirFlip bool
+	// the minimum power that can be set for the motor to prevent stalls
+	minPowerPct float64
+	// the maximum power that can be set for the motor
+	maxPowerPct float64
+	// the freewheel RPM of the motor
+	maxRPM float64
+
+	// A manager to ensure only a single operation is happening at any given time since commands could overlap on the serial port
+	opMgr operation.SingleOperationManager
+}
+
+// Config adds DimensionEngineering-specific config options.
+type Config struct {
+	// path to /dev/ttyXXXX file
+	SerialPath string `json:"serial_path"`
+
+	// The baud rate of the controller
+	BaudRate int `json:"serial_baud_rate,omitempty"`
```

Comment:
do you ever use this field? it looks like you always constrain the baudrate to 9600